src/PHPCR/NodeInterface.php

```diff
@@ -485,9 +487,8 @@ public function getNode($relPath);
      * @param string|array $typeFilter a filter or an array of filters for the
      *      node type names to find.
      *
-     * @return Iterator over all (matching) child Nodes implementing
-     *      <b>SeekableIterator</b> and <b>Countable</b>. Keys are the Node
-     *      names, values the corresponding NodeInterface instances.
+     * @return Iterator<string, NodeTypeInterface> over all (matching) child Nodes implementing <b>SeekableIterator</b>
+     *                                             and <b>Countable</b>. Keys are the Node names.
```

Comment:
`NodeTypeInterface` looks strange for me here should this not be `static` or `self`?

e.g.:

<img width="762" alt="Bildschirmfoto 2023-09-02 um 14 19 23" src="https://github.com/phpcr/phpcr/assets/1698337/69428f73-2664-4685-a6a7-b136ac6a960e">


https://github.com/sulu/sulu/blob/654681b446ee3d4ac4c6343ecc8f39846dc15e17/src/Sulu/Bundle/PageBundle/Document/Subscriber/PublishSubscriber.php#L162

Reply:
At sulu it would fix some of this issues:

<img width="541" alt="Bildschirmfoto 2023-09-02 um 14 28 52" src="https://github.com/phpcr/phpcr/assets/1698337/cfc13309-e27f-44e3-81b2-f1304abb5f22">